### PR TITLE
feat: activate prague always

### PR DESCRIPTION
**Pull request by onbjerg**
Adds a custom chainspec parser that just inserts Prague as a hardfork from timestamp 0, always. This logic should be replaced with a custom hardfork name that we parse and map to Prague, once that is possible.

This can be reviewed as its own thing, but requires #111

### CODE REVIEW

File: crates/node/src/chainspec.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+pub struct AlphanetChainSpecParser;
+
+impl ChainSpecParser for AlphanetChainSpecParser {
+    type ChainSpec = ChainSpec;
+
+    const SUPPORTED_CHAINS: &'static [&'static str] = &["dev"];
+
+    fn parse(s: &str) -> eyre::Result<Arc<Self::ChainSpec>> {
+        Ok(match s {
+            "dev" => ALPHANET_DEV.clone(),
```
Comment (mattsse) [inline crates/node/src/chainspec.rs:70]: this lacks "alphanet" and default spec?
don't think we should use dev as the default chain

File: crates/node/src/chainspec.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            s => {
+                let mut chainspec = parse_custom_chain_spec(s)?;
+
+                // NOTE(onbjerg): This is a temporary workaround until we figure out a better way to
+                // activate Prague based on a custom fork name. Currently there does not seem to be
+                // a good way to do it.
+                chainspec.hardforks.insert(EthereumHardfork::Prague, ForkCondition::Timestamp(0));
```
Comment (mattsse) [inline crates/node/src/chainspec.rs:77]: this should work, but can we please add a test here

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/node/src/chainspec.rs (hunk 2 above)
Comment (onbjerg) [reply crates/node/src/chainspec.rs:77]: yes as mentioned on the meeting yesterday i am adding a test

File: crates/node/src/chainspec.rs (hunk 1 above)
Comment (onbjerg) [reply crates/node/src/chainspec.rs:70]: what is the difference between "alphanet" and "default spec"

Comment (mattsse) [reply crates/node/src/chainspec.rs:70]: default supported chain spec, currently there's only dev which is used as the default when parsing.

Comment (onbjerg) [reply crates/node/src/chainspec.rs:70]: okay, so you just meant that there is one spec missing

Comment (mattsse) [reply crates/node/src/chainspec.rs:70]: yeah, there's no alphanet spec, there's only dev
so --chain alphanet would not work.

File: crates/node/src/evm.rs [hunk 3]
```diff
@@ -28,7 +28,7 @@ use reth_revm::{
 use std::sync::Arc;
 
 /// Custom EVM configuration
-#[derive(Debug, Clone, Default)]
```
Comment (onbjerg) [inline crates/node/src/evm.rs:31]: removed default so we don't accidentally use it and end up with a bogus chainspec

**Comment by onbjerg** [comment]
Rfr, this does work, however the payload building is a bit scuffed until https://github.com/paradigmxyz/reth/pull/10510 is solved. For that reason I also won't add an e2e test in this PR since it would just fail

**Pull request merged**